main/src/ca/uwaterloo/flix/language/phase/MonoDefs.scala

```diff
@@ -89,8 +89,6 @@ object MonoDefs {
             case Some(tpe) => tpe.map(default)
             case None => default(t)
           }
-          // Erase concrete effects like Print.
-          case Type.Cst(TypeConstructor.Effect(_), _) => Type.EffUniv
```

Comment:
So presumably now we create specific versions for every demanded effectset-- which is probably okay. Later we can simplify that to Pure, Impure, ControlImpure.